src/Build/Logging/FileLogger.cs

```diff
@@ -86,6 +86,15 @@ private void InitializeFileLogger(IEventSource eventSource, int nodeCount)
 
             try
             {
+                try
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(Path.GetFullPath(_logFileName)));
```

Comment:
Hmm. I see what Jeff was going for here, and I like the idea of isolating a failure of the path-manipulation stuff, but I don't like throwing away possibly-useful information like "you don't have permission to create that folder" that could be thrown from `CreateDirectory()`. Maybe something like:

```C#
string logDirectory = null;
try
{
    logDirectory = Path.GetDirectoryName(Path.GetFullPath(_logFileName));
}
catch
{
    // Directory creation is best-effort; if finding its path fails don't create the directory
    // and possibly let OpenWrite() below report the failure
}

if (logDirectory != null)
{
    Directory.CreateDirectory(logDirectory);
}
```

How's that sound?